docs/devguide/devguide.md

```diff
@@ -98,19 +100,24 @@ To properly track changes across several Modules, you need to follow the recipe.
 ### Recipe
 Split up the data flow for each direction, one call for reading only, one call for writing only.
 Keep in mind that the caller by definition is "left" in the module graph and the callee is "right". The callee is the end of a callback, but for this pattern this has nothing to do with the direction the data flows in, which results in the rules defined below.
+
+![bidir example](bidir.png)
+
 - set up a ```uint32_t version_DATA``` in each Module
```

Comment:
Neither the consumer nor the producer should have their own version. This is a property of the DATA and should therefore always be "owned" by the holder. Imagine two producers A and B. A updates DATA before B. If B sets DATA with the version stored in B, it can happen that the set_version is lower than the get_version and hasUpdate() breaks.